exporter/exporter/main.py

```diff
@@ -0,0 +1,85 @@
+import os
+import json
+from log import log
+from dotenv import load_dotenv
+from db import setup_database_connection
+from pydantic import BaseModel
+from kafka import KafkaConsumer, admin,TopicPartition
+from exporter import export_data
+
+# load the env file into the environment
+load_dotenv()
+
+# Settings
+KAFKA_TOPIC=os.getenv('KAFKA_TOPIC', 'vlab-exports')
+KAFKA_SERVERS=os.getenv('KAFKA_SERVERS', '').split(',')
+KAFKA_GROUP_ID=os.getenv('KAFKA_GROUP_ID', 'exporter')
+DATABASE_URL=os.getenv("DATABASE_URL")
+
+class KafkaMessage(BaseModel):
+    event: str
+    survey: str
+    user: str
+    options: dict
+
+def app():
+    """
+    the entrypoint to the application
+    """
+
+    # Setup consumer
+    consumer = setup_kafka_consumer()
+
+    # setup database connection
+    conn = setup_database_connection(DATABASE_URL)
+
+    log.info("ready to start receiving messages")
+    # Run connection using with to handle termination to database properly
+    with conn:
+        for message in consumer:
+            try:
+                if not message.value:
+                    continue
+                process(conn, message.value)
+            # catch all uncaught exceptions and 
+            # print out error
+            except Exception as e:
```

Comment:
I would maybe kill the whole process on uncaught, or unexpected, errors. Then at least we get notified and can handle them - also if something really funky is happening, killing it all and letting k8s restart is always a good thing.